### DISCUSSION TITLE
Search/filter

### DISCUSSION THREAD

**Discussion by pguptaDS** [twilio-labs/paste#2255] [Q&A] [answered]
I remembering seeing a filter/search pattern figma somewhere but can't find it anywhere. I wanted to ask what are the recommendations for filter/search on a table now? This is what I had [designed](https://www.figma.com/file/fOndYlhuIYuMU4okv2lRyb/RBAC-for-Flex?node-id=435%3A109804) based on the paste team's figma file but my engineering team raised concerns that not having a specific search button and relying solely on the search icon might be problematic. Thoughts?

**Comment by shleewhite** [comment, marked as answer]
Hi @pguptaDS! 

There is a [Filter Pattern spec document](https://docs.google.com/document/d/1w7_BIeFNlg3b8Nm_r-2WlqTR7MQyhcI5vFmF_rM-pnQ/edit) and [a figma file](https://www.figma.com/file/S4z0Kqjb9AYosnkRQldLte/branch/rNbsDOvtGr64Yl8o9Cwy0j/Paste-Patterns?node-id=1391%3A47182). Also, we have a ticket right now to add the pattern to the website, so that should be published soon!

Can you elaborate on their concerns? In the spec, there is an "Apply" button for the group of filters and I believe for the search input, the table is supposed to filter as the user types.
